File: 34.在排序数组中查找元素的第一个和最后一个位置.go
/*
 * @lc app=leetcode.cn id=34 lang=golang
 *
 * [34] 在排序数组中查找元素的第一个和最后一个位置
 */

// @lc code=start
func searchRange(nums []int, target int) []int {
	n := len(nums)
	start := sort.SearchInts(nums, target)

	if start == n || nums[start] != target {
		return []int{-1, -1}
	}

	end := sort.SearchInts(nums, target+1)

	return []int{start, end - 1}
}

// @lc code=end

File: 34.在排序数组中查找元素的第一个和最后一个位置.py
#
# @lc app=leetcode.cn id=34 lang=python
#
# [34] 在排序数组中查找元素的第一个和最后一个位置
#

# @lc code=start
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # nums[left-1] < target <= nums[left]
        left = bisect_left(nums, target)
        if left == len(nums) or nums[left] != target:
            return [-1, -1]
        # nums[right-1] <= target < nums[right]
        right = bisect_right(nums, target)
        return [left, right-1]
# @lc code=end

